### PR TITLE
Upgrade hyper/mime/url

### LINKED ISSUE #318: Upgrade to Hyper 0.8
I'd like to add support for Nickel in [multipart](https://github.com/cybergeek94/multipart) but we're currently on Hyper 0.7 over there, with [some pressure to upgrade to 0.8](https://github.com/LukasKalbertodt/telegram-bot/pull/34#issuecomment-199257644). Is there a reason you guys are so far behind?

### PR DESCRIPTION
**Pull request by cybergeek94**
The `examples/template.rs` example fails but it says it's deprecated? Anyways, the assertion fails because the line endings changed, probably because I'm on Windows and Git checks out with CRLF line endings by default.

```
failures:

---- examples::template::renders_data stdout ----
        Parsed: port=58433 from "Listening on http://[::1]:58433"
thread 'examples::template::renders_data' panicked at 'assertion failed: `(left == right)` (left: `"<html>\r\n
   <head>\r\n        <title>\r\n            nickel.rs - example\r\n        </title>\r\n    </head>\r\n    <body>\r\n    <h1>\r\n        Hello user!\r\n    </h1>\r\n    </body>\r\n</html>"`, right: `"<html>\n    <head>\n
     <title>\n            nickel.rs - example\n        </title>\n    </head>\n    <body>\n    <h1>\n        Hello user!\n    </h1>\n    </body>\n</html>"`)', tests\examples\template.rs:23
Unparsed Stdout:
```

Edit: forgot to add this:

Closes #318

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were style issues with your Pull Request:

 * Commit: d1168a97eafd5716e4fe437f8f08e9072c9f062c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/nickel-org/nickel.rs/blob/master/contributing.md

---
This message was auto-generated by https://gitcop.com

**Comment by cybergeek94** [comment]
I can fix the commit message but the link to the conventions document from `CONTRIBUTING.md` is broken.

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 
 name = "nickel"
-version = "0.7.3"
+version = "0.8.0"
```
Comment (Ryman) [inline Cargo.toml]: Could you leave the version bump out of this commit? I'd prefer if we bumped it alongside a changelog update.

**Comment by Ryman** [comment]
@cybergeek94 Ah, thanks for the heads up, I'll fix the link, it looks like they've moved repos to [here](https://github.com/stevemao/conventional-changelog-angular/blob/master/convention.md).

**Comment by Ryman** [comment]
Looks like we won't be able to support 1.3 any more since hyper's now forcing the timeouts feature, and one of our dependencies is using `Result::expect`. So it seems 1.4 is the new minimum.

**Comment by cybergeek94** [comment]
What would be the `type` and `scope` of this commit? I guess it would be a breaking change as well.

**Comment by Ryman** [comment]
`feat(*): bump hyper, mime and url dependency versions` would be fine, and you're right it is a breaking change :)

**Comment by cybergeek94** [comment]
@Ryman amended, force-pushed.

**Comment by cybergeek94** [comment]
Ah, still needs to upgrade to a minimum of Rust 1.4 as you said. Do you want me to do that here? In a new commit or amend the current one?

**Comment by Ryman** [comment]
@cybergeek94 A new commit sounds fine, can add a basic description of why in the description :+1:

**Comment by cybergeek94** [comment]
@Ryman how's that look?

**Comment by cybergeek94** [comment]
Crap, looks like we might need to go even newer than that. The `iter_order` feature was stabilized in 1.5.

**Comment by Ryman** [comment]
Ah, that's a shame but I'm happy enough to move to it if that's what significant dependencies (`url`) have moved to. cc @cburgdorf @SimonPersson

**Comment by Ryman** [comment]
Might want to adjust the message as it's not really hyper's fault, more just the general ecosystem has moved on :P

**Comment by cybergeek94** [comment]
Okay, let's see if that works.

**Comment by cybergeek94** [comment]
Passed this time, though there's some failed tests on nightly (which Travis allows, so not sure that's a problem?)

**Comment by Ryman** [comment]
Merged as part of #321, thanks for your contribution!

**Pull request closed without merging**